helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -410,10 +433,78 @@ public void close() {
   }
 
   /**
+   * Undeploy jobs left over by {@link TemporaryJobs}. TemporaryJobs should clean these up,
+   * but sometimes a few are left behind for whatever reason.
+   */
+  private void undeployLeftoverJobs() {
+    try {
+      // List all jobs. TemporaryJobs should delete jobs in addition to undeploying them.
+      // So any jobs found at this point have only been partially cleaned up.
+      final Map<JobId, Job> jobs = heliosClient.jobs().get();
+      if (jobs.size() > 0) {
+        log.info("There are jobs in helios-solo left over by TemporaryJobs. " +
+                 "Undeploying and deleting them now. Jobs: {}", jobs.keySet());
+      }
+
+      for (final JobId jobId : jobs.keySet()) {
+        final JobStatus jobStatus = heliosClient.jobStatus(jobId).get();
+        final Map<String, TaskStatus> statuses = jobStatus.getTaskStatuses();
+
+        for (Map.Entry<String, TaskStatus> status : statuses.entrySet()) {
+          final String host = status.getKey();
+          if (status.getValue().getGoal().equals(Goal.START)) {
+            log.info("Job {} is still set to START on host {}. Undeploying it now.", jobId, host);
+            final JobUndeployResponse undeployResponse = heliosClient.undeploy(jobId, host).get();
+            log.info("Undeploy response for job {} is {}.", jobId, undeployResponse.getStatus());
+
+            log.info("Waiting for job {} to actually be undeployed...", jobId);
+            awaitJobUndeployed(heliosClient, host, jobId, 400, TimeUnit.SECONDS);
```

Comment:
I think it would be nice if this was customizable in any case, for instance when constructing the HeliosSoloDeployment instance (there are other builder args already)